packages/form-core/src/util-types.ts

```diff
@@ -83,17 +85,40 @@ type PrefixFromDepth<
   TDepth extends any[],
 > = TDepth['length'] extends 0 ? T : `.${T}`
 
-export type DeepValue<TValue, TAccessor> = TValue extends Record<
-  string | number,
-  any
->
-  ? TAccessor extends `${infer TBefore}[${infer TBrackets}].${infer TAfter}`
-    ? DeepValue<TValue[TBefore][TBrackets], TAfter>
-    : TAccessor extends `[${infer TBrackets}]`
-      ? DeepValue<TValue, TBrackets>
-      : TAccessor extends `${infer TBefore}[${infer TBrackets}]`
-        ? DeepValue<TValue[TBefore], TBrackets>
-        : TAccessor extends `${infer TBefore}.${infer TAfter}`
-          ? DeepValue<TValue[TBefore], TAfter>
-          : TValue[TAccessor & string]
-  : never
+export type DeepValue<
```

Comment:
I don't know if it's an issue we can solve, but this type is pretty complex and hard to understand (for me)... 😅 

TS in libs can get pretty crazy, I get that, but I think `DeepValue` has the potential to make it very hard to add new features or fix bugs in the future. And `DeepValue` is a pretty foundational part of the code base.

_How could we address this issue?_ It's not entirely fixable, but maybe we could add comments to make the intent clearer in some places. E.g.:

```ts
// Limit the depth of recursion to 3 levels
TDepth['length'] extends 3
  ? never
  : TValue extends ReadonlyArray<any> ? ...
```

It'd be also good to document the type in some way, like:
```ts
/**
   Infer the type of a deeply nested property within an object or array.
   See the tests in `util-types.test-d.ts` for usage.
*/
export type DeepValue<TValue, TAccessor, TDepth extends any[] = []>
```

In general, I'm not a fan of commenting code, because comments tend to get outdated rather quickly, but in this case, I think they would help us immensely, and we do proper code reviews anyways.

Here's my proposal: if you're okay with adding comments, I would be happy to do this work in a suggestion or commit to this PR, and then we could discuss them.

There's also the possibility of making smaller types for the different logical groups in the type (e.g.: one branch of a conditional infers the property from a `Record` (could be `DeepValueInRecord`??), the other one from an `Array` (`DeepValueInArray`??), and things like that). I feel like this approach could overcomplicate things, so I'd avoid it (this is just a gut feeling, so maybe I'm wrong).

Wdyt?

Reply:
Mentioned this privately in our maintainers channel but want to confirm publicly; **Comments are always welcome in the codebase!**

In fact, I'm glad you're advocating for this; it'll a great change for our users and future collaborators.